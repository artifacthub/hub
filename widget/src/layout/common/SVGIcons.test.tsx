import { render, screen } from '@testing-library/react';

import SVGIcons from './SVGIcons';

describe('SVGIcons', () => {
  it('creates snapshot', () => {
    const { asFragment } = render(<SVGIcons name="logo" />);
    expect(asFragment).toMatchSnapshot();
  });

  it('renders logo', () => {
    render(<SVGIcons name="logo" />);
    expect(screen.getByTitle('logo'));
  });

  it('renders stars', () => {
    render(<SVGIcons name="stars" />);
    expect(screen.getByTitle('stars'));
  });

  it('renders verified', () => {
    render(<SVGIcons name="verified" />);
    expect(screen.getByTitle('verified'));
  });

  it('renders official', () => {
    render(<SVGIcons name="official" />);
    expect(screen.getByTitle('official'));
  });

  it('renders helm icon', () => {
    render(<SVGIcons name="helm" />);
    expect(screen.getByTitle('helm'));
  });

  it('renders helm-plugin icon', () => {
    render(<SVGIcons name="helm-plugin" />);
    expect(screen.getByTitle('helm-plugin'));
  });

  it('renders falco icon', () => {
    render(<SVGIcons name="falco" />);
    expect(screen.getByTitle('falco'));
  });

  it('renders opa icon', () => {
    render(<SVGIcons name="opa" />);
    expect(screen.getByTitle('opa'));
  });

  it('renders tinkerbell icon', () => {
    render(<SVGIcons name="tinkerbell" />);
    expect(screen.getByTitle('tinkerbell'));
  });

  it('renders tekton icon', () => {
    render(<SVGIcons name="tekton" />);
    expect(screen.getByTitle('tekton'));
  });

  it('renders olm icon', () => {
    render(<SVGIcons name="olm" />);
    expect(screen.getByTitle('olm'));
  });

  it('renders krew icon', () => {
    render(<SVGIcons name="krew" />);
    expect(screen.getByTitle('krew'));
  });

  it('renders keda icon', () => {
    render(<SVGIcons name="keda" />);
    expect(screen.getByTitle('keda'));
  });

  it('renders coredns icon', () => {
    render(<SVGIcons name="coredns" />);
    expect(screen.getByTitle('coredns'));
  });

  it('renders keptn icon', () => {
    render(<SVGIcons name="keptn" />);
    expect(screen.getByTitle('keptn'));
  });

  it('renders container icon', () => {
    render(<SVGIcons name="container" />);
    expect(screen.getByTitle('container'));
  });

  it('renders kubewarden icon', () => {
    render(<SVGIcons name="kubewarden" />);
    expect(screen.getByTitle('kubewarden'));
  });

  it('does not render when name is not in the list', () => {
    render(<SVGIcons name="not-listed" />);
    expect(screen.getByTestId('iconWrapper')).toBeEmptyDOMElement();
  });
});
