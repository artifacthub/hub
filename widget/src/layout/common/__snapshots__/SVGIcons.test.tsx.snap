// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGIcons creates snapshot 1`] = `[Function]`;
